src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +57,16 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.concat(
+                IntStream.range(0, numbers.size())
+                        .filter(i -> i % 2 != 0)
+                        .map(i -> numbers.get(i) - 1),
+                IntStream.range(0, numbers.size())
+                        .filter(i -> i % 2 == 0)
+                        .map(numbers::get))
```

Comment:
can we do this with one int stream but with conditional mapping?

Reply:
done